airflow/migrations/versions/0050_3_0_0_remove_pickled_data_from_dagrun_table.py

```diff
@@ -0,0 +1,221 @@
+    SELECT dag_id, run_id, queued_at, logical_date, start_date, end_date, state, creating_job_id, external_trigger, run_type, triggered_by, conf, data_interval_start, data_interval_end, last_scheduling_decision, log_template_id, updated_at, clear_number, backfill_id, dag_version_id
+    FROM  dag_run
+    WHERE conf IS NOT NULL AND {condition};
+
+                """
+        )
+    )
+
+    # Delete the pickled data from the dag_run table so that we can update the column type
+    conn.execute(text(f"DELETE FROM dag_run WHERE conf IS NOT NULL AND {condition}"))
+
+    # Update the conf column type to JSON
+    if dialect == "postgresql":
+        op.execute(
+            """
+            ALTER TABLE dag_run
+            ALTER COLUMN conf TYPE JSONB
+            USING CASE
+                WHEN conf IS NOT NULL THEN CAST(CONVERT_FROM(conf, 'UTF8') AS JSONB)
+                ELSE NULL
+            END
+            """
+        )
+    elif dialect == "mysql":
+        op.add_column("dag_run", sa.Column("conf_json", sa.JSON(), nullable=True))
+        op.execute("UPDATE dag_run SET conf_json = CAST(conf AS CHAR CHARACTER SET utf8mb4)")
+        op.drop_column("dag_run", "conf")
+        op.alter_column("dag_run", "conf_json", existing_type=sa.JSON(), new_column_name="conf")
+    elif dialect == "sqlite":
+        # Rename the existing `conf` column to `conf_old`
+        with op.batch_alter_table("dag_run", schema=None) as batch_op:
+            batch_op.alter_column("conf", new_column_name="conf_old")
+
+        # Add the new `conf` column with JSON type
+        with op.batch_alter_table("dag_run", schema=None) as batch_op:
+            batch_op.add_column(sa.Column("conf", sa.JSON(), nullable=True))
+
+        # Migrate data from `conf_old` to `conf`
+        conn.execute(
+            text(
+                """
+                UPDATE dag_run
+                SET conf = json(CAST(conf_old AS TEXT))
+                WHERE conf_old IS NOT NULL
+                """
+            )
+        )
+
+        # Drop the old `conf_old` column
+        with op.batch_alter_table("dag_run", schema=None) as batch_op:
+            batch_op.drop_column("conf_old")
+
+
+def downgrade():
+    """Unapply Remove pickled data from dagrun table."""
+    conn = op.get_bind()
+    dialect = conn.dialect.name
+    inspector = sa.inspect(conn)
+
+    insert_query = text(
+        """
+                       INSERT INTO dag_run (dag_id, run_id, queued_at, logical_date, start_date, end_date, state, creating_job_id, external_trigger, run_type, triggered_by, conf, data_interval_start, data_interval_end, last_scheduling_decision, log_template_id, updated_at, clear_number, backfill_id, dag_version_id)
```

Comment:
This doesn't feel right - what if I upgraded 2 weeks ago and have since done a db clean?!

Reply:
This will only [run](https://github.com/apache/airflow/blob/2b6abf740eb365fb126145c532a08774eb7d6f31/airflow/migrations/versions/0050_3_0_0_remove_pickled_data_from_dagrun_table.py#L187) if `_dag_run_archive` exits.

Reply:
Removed this